sling.go

```diff
@@ -497,14 +485,23 @@ func (s *Sling) doDecode(req *http.Request, successV, failureV interface{}) (*ht
 	// reuse HTTP/1.x "keep-alive" TCP connections if the Body is
 	// not read to completion and closed.
 	// See: https://golang.org/pkg/net/http/#Response
-	defer io.Copy(ioutil.Discard, resp.Body)
+	defer io.Copy(io.Discard, resp.Body)
+
+	// Don't try to decode on 204s
+	if resp.StatusCode == http.StatusNoContent {
+		return resp, nil
+	}
+
+	// Don't decode if the content length is 0
+	if resp.ContentLength == 0 {
+		if failureV == nil && !isSuccessful(resp.StatusCode) {
+			return resp, fmt.Errorf("status code %d was not successful and had no body", resp.StatusCode)
+		}
```

Comment:
In our tests, there's a few cases where there's a Content-Length of 0 and since we used to check the status code by hand it was expected to return an error. I think that behavior makes more sense so I rolled it into sling here.

Reply:
Sounds good.